freqtrade/plugins/pairlist/MarketCapPairList.py

```diff
@@ -0,0 +1,182 @@
+                "help": "How to interpret the number",
+            },
+            "refresh_period": {
+                "type": "number",
+                "default": 86400,
+                "description": "Refresh period",
+                "help": "Refresh period in seconds",
+            }
+        }
+
+    def gen_pairlist(self, tickers: Tickers) -> List[str]:
+        """
+        Generate the pairlist
+        :param tickers: Tickers (from exchange.get_tickers). May be cached.
+        :return: List of pairs
+        """
+        # Generate dynamic whitelist
+        # Must always run if this pairlist is not the first in the list.
+        pairlist = self._marketcap_cache.get('pairlist_mc')
+        if pairlist:
+            # Item found - no refresh necessary
+            return pairlist.copy()
+        else:
+            # Use fresh pairlist
+            # Check if pair quote currency equals to the stake currency.
+            _pairlist = [k for k in self._exchange.get_markets(
+                quote_currencies=[self._stake_currency],
+                tradable_only=True, active_only=True).keys()]
+            # No point in testing for blacklisted pairs...
+            _pairlist = self.verify_blacklist(_pairlist, logger.info)
+
+            pairlist = self.filter_pairlist(_pairlist, tickers)
+            self._marketcap_cache['pairlist_mc'] = pairlist.copy()
+
+        return pairlist
+
+    def filter_pairlist(self, pairlist: List[str], tickers: Dict) -> List[str]:
+        """
+        Filters and sorts pairlist and returns the whitelist again.
+        Called on each bot iteration - please use internal caching if necessary
+        :param pairlist: pairlist to filter or sort
+        :param tickers: Tickers (from exchange.get_tickers). May be cached.
+        :return: new whitelist
+        """
+        marketcap_list = self._marketcap_cache.get('marketcap')
+        can_filter = False
+
+        if marketcap_list:
+            can_filter = True
+        else:
+            data = self._coingekko.get_coins_markets(vs_currency='usd', order='market_cap_desc',
+                                                     per_page='250', page='1', sparkline='false',
+                                                     locale='en')
+            if data:
+                marketcap_list = [row['symbol'] for row in data]
+                self._marketcap_cache['marketcap'] = marketcap_list
+                can_filter = True
+
+        if can_filter:
+            filtered_pairlist = []
+
+            if self._mode == 'top_rank':
+                top_marketcap = marketcap_list[:self._number_assets:]
+
+                for pair in pairlist:
+                    base = pair.split('/')[0]
+                    if base.lower() in top_marketcap:
+                        filtered_pairlist.append(pair)
+                    else:
+                        logger.info(f"Remove {pair} from whitelist because it's not ranked "
```

Comment:
should use `self.log_once(` ... to avoid logging on every iteration in some odd cases.